src/Autofac/Core/Activators/Reflection/ReflectionActivator.cs

```diff
@@ -147,7 +213,8 @@ private object ActivateInstance(IComponentContext context, IEnumerable<Parameter
     private BoundConstructor[] GetAllBindings(ConstructorBinder[] availableConstructors, IComponentContext context, IEnumerable<Parameter> parameters)
     {
         // Most often, there will be no `parameters` and/or no `_defaultParameters`; in both of those cases we can avoid allocating.
-        var prioritisedParameters = parameters.Any() ? EnumerateParameters(parameters) : _defaultParameters;
+        // Do a reference compare with the NoParameters constant; faster than an Any() check for the common case.
```

Comment:
I think this comment goes down in the `GetAllParameters` method since the associated code moved.